framework/ldtest/filter.go

```diff
@@ -2,12 +2,25 @@ package ldtest
 
 import (
 	"fmt"
+	"io"
 	"regexp"
 	"strings"
 )
 
-// Filter is a function that can determine whether to run a specific test or not.
-type Filter func(TestID) bool
+// Filter is an object that can determine whether to run a specific test or not.
+type Filter interface {
+	Match(id TestID) bool
+}
```

Comment:
This doesn't have much to do with client-side testing, it's just that I realized this API was a bit inconvenient when I started refactoring the top-level entry points.